e2e/tests/upgrades/upgrade_test.go

```diff
@@ -81,9 +83,89 @@ func (s *UpgradeTestSuite) TestV4ToV5ChainUpgrade() {
 
 	ctx := context.Background()
 
-	s.SetupChainsRelayerAndChannel(ctx)
-	chainA, _ := s.GetChains()
+	relayer, channelA := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
 
-	chainAUser := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
-	s.UpgradeChain(ctx, chainA, chainAUser, "normal upgrade", targetVersion)
+	chainADenom := chainA.Config().Denom
+
+	// create separate users specifically for the upgrade proposal to more easily verify starting
+	// and end balances of chainA and chainB users.
+	chainAUpgradeProposalWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	chainBUpgradeProposalWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	chainAAddress := chainAWallet.Bech32Address(chainA.Config().Bech32Prefix)
+
+	chainBWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+	chainBAddress := chainBWallet.Bech32Address(chainB.Config().Bech32Prefix)
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 1, chainA, chainB), "failed to wait for blocks")
+
+	t.Run("native IBC token transfer from chainA to chainB, sender is source of tokens", func(t *testing.T) {
+		transferTxResp, err := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(transferTxResp)
+	})
+
+	t.Run("tokens are escrowed", func(t *testing.T) {
+		actualBalance, err := s.GetChainANativeBalance(ctx, chainAWallet)
+		s.Require().NoError(err)
+
+		expected := testvalues.StartingTokenAmount - testvalues.IBCTransferAmount
+		s.Require().Equal(expected, actualBalance)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	chainBIBCToken := testsuite.GetIBCToken(chainADenom, channelA.Counterparty.PortID, channelA.Counterparty.ChannelID)
+
+	t.Run("packets are relayed", func(t *testing.T) {
+		s.AssertPacketRelayed(ctx, chainA, channelA.PortID, channelA.ChannelID, 1)
+
+		actualBalance, err := chainB.GetBalance(ctx, chainBAddress, chainBIBCToken.IBCDenom())
+		s.Require().NoError(err)
+
+		expected := testvalues.IBCTransferAmount
+		s.Require().Equal(expected, actualBalance)
+	})
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 5, chainA, chainB), "failed to wait for blocks")
+
+	t.Run("upgrade both chains", func(t *testing.T) {
+		var eg errgroup.Group
+		eg.Go(func() error {
+			s.UpgradeChain(ctx, chainA, chainAUpgradeProposalWallet, "normal upgrade", targetVersion)
+			return nil
+		})
+		eg.Go(func() error {
+			s.UpgradeChain(ctx, chainB, chainBUpgradeProposalWallet, "normal upgrade", targetVersion)
+			return nil
+		})
```

Comment:
I think it makes more sense to only upgrade chainA and send transfers to/from chainA. Otherwise you need to ensure you pick a binary that can upgrade to the targetVersion (not true for cross compatibility where chainB is on a higher version)

Reply:
ah yeah that makes sense 👍